src/main/java/community/whatever/onembackendjava/controller/UrlShortenController.java

```diff
@@ -0,0 +1,50 @@
+package community.whatever.onembackendjava.controller;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import community.whatever.onembackendjava.dto.req.ShortenedURLCreateRequest;
+import community.whatever.onembackendjava.dto.res.ShortenedURLCreateResponse;
+import community.whatever.onembackendjava.service.ShortenURLService;
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequiredArgsConstructor
+public class UrlShortenController {
+
+	private final ShortenURLService service;
+
+	@Deprecated
+	@PostMapping("/shorten-url/search")
+	public String shortenUrlSearch(@RequestBody String key) {
+		String originURL = service.getOriginURL(key);
+		return originURL;
+
+	}
```

Comment:
```suggestion
        /**
	 * @deprecated use {@link #getOriginalUrl(String)}
	 */
	@Deprecated
	@PostMapping("/shorten-url/search")
	public String shortenUrlSearch(@RequestBody String key) {
		String originURL = service.getOriginURL(key);
		return originURL;
	}
```

`nit` Deprecated 사용 좋습니다. 좀 더 친절하게 하려면 javadoc 도 쓸 수 있습니다. (사실 이정도까지는 많이 안함)

Reply:
나눠서 커밋해야할까요?

Reply:
멘티의 선택 입니다.

--- 

저는
- 상황에 따라 포스 푸쉬를 하기도 합니다.
- 실무에서는 pr 단위의 squash merge 하기 때문에 하나의 pr 에서 commit 이 늘어나는 것은 신경쓰지 않습니다.